File: tmp/fix_search.js
const fs = require('fs');

let content = fs.readFileSync('search.html', 'utf-8');

const target1 = `                    // Fallback to RAW_JSON if DB is empty\r
                    funeralHomesData = (typeof RAW_JSON !== 'undefined' && RAW_JSON.business) ? RAW_JSON.business : [];`;

const replace1 = `                    // Fallback to RAW_JSON if DB is empty\r
                    if (typeof RAW_JSON !== 'undefined' && RAW_JSON.business) {\r
                        funeralHomesData = RAW_JSON.business;\r
                    } else if (typeof REAL_DATA !== 'undefined') {\r
                        funeralHomesData = Object.keys(REAL_DATA).map((name, index) => {\r
                            const info = REAL_DATA[name];\r
                            return {\r
                                id: 'fallback-' + index,\r
                                name: name,\r
                                address: info.address || '',\r
                                latitude: info.latitude || 37.5665,\r
                                longitude: info.longitude || 126.9780,\r
                                image_url: info.imageUrl || '',\r
                                is_alliance: true,\r
                                prices: info.prices || []\r
                            };\r
                        });\r
                    } else {\r
                        funeralHomesData = [];\r
                    }`;

const target2 = `                // Fallback\r
                funeralHomesData = (typeof RAW_JSON !== 'undefined' && RAW_JSON.business) ? RAW_JSON.business : [];`;

const replace2 = `                // Fallback\r
                if (typeof RAW_JSON !== 'undefined' && RAW_JSON.business) {\r
                    funeralHomesData = RAW_JSON.business;\r
                } else if (typeof REAL_DATA !== 'undefined') {\r
                    funeralHomesData = Object.keys(REAL_DATA).map((name, index) => {\r
                        const info = REAL_DATA[name];\r
                        return {\r
                            id: 'fallback-' + index,\r
                            name: name,\r
                            address: info.address || '',\r
                            latitude: info.latitude || 37.5665,\r
                            longitude: info.longitude || 126.9780,\r
                            image_url: info.imageUrl || '',\r
                            is_alliance: true,\r
                            prices: info.prices || []\r
                        };\r
                    });\r
                } else {\r
                    funeralHomesData = [];\r
                }`;

if (content.includes(target1)) {
    content = content.replace(target1, replace1);
    console.log("Replaced target1");
} else {
    console.log("Could not find target1");
}

if (content.includes(target2)) {
    content = content.replace(target2, replace2);
    console.log("Replaced target2");
} else {
    console.log("Could not find target2");
}

fs.writeFileSync('search.html', content, 'utf-8');
console.log("File saved.");

File: tmp/fix_search.py
import codecs

with codecs.open('search.html', 'r', 'utf-8') as f:
    content = f.read()

target = """                // Fetch from Supabase
                const data = await Auth.getAllFuneralHomes();
                if (data && data.length > 0) {
                    funeralHomesData = data;
                } else {
                    // Fallback to RAW_JSON if DB is empty
                    funeralHomesData = (typeof RAW_JSON !== 'undefined' && RAW_JSON.business) ? RAW_JSON.business : [];
                }"""

replacement = """                // Fetch from Supabase
                const data = await Auth.getAllFuneralHomes();
                if (data && data.length > 0) {
                    funeralHomesData = data;
                } else {
                    // Fallback to RAW_JSON if DB is empty
                    if (typeof RAW_JSON !== 'undefined' && RAW_JSON.business) {
                        funeralHomesData = RAW_JSON.business;
                    } else if (typeof REAL_DATA !== 'undefined') {
                        // Fallback to REAL_DATA
                        funeralHomesData = Object.keys(REAL_DATA).map((name, index) => {
                            const info = REAL_DATA[name];
                            return {
                                id: 'fallback-' + index,
                                name: name,
                                address: info.address || '',
                                latitude: info.latitude || 37.5665,
                                longitude: info.longitude || 126.9780,
                                image_url: info.imageUrl || '',
                                is_alliance: true,
                                prices: info.prices || []
                            };
                        });
                    } else {
                        funeralHomesData = [];
                    }
                }"""

target2 = """                // Fallback
                funeralHomesData = (typeof RAW_JSON !== 'undefined' && RAW_JSON.business) ? RAW_JSON.business : [];"""

replacement2 = """                // Fallback
                if (typeof RAW_JSON !== 'undefined' && RAW_JSON.business) {
                    funeralHomesData = RAW_JSON.business;
                } else if (typeof REAL_DATA !== 'undefined') {
                    funeralHomesData = Object.keys(REAL_DATA).map((name, index) => {
                        const info = REAL_DATA[name];
                        return {
                            id: 'fallback-' + index,
                            name: name,
                            address: info.address || '',
                            latitude: info.latitude || 37.5665,
                            longitude: info.longitude || 126.9780,
                            image_url: info.imageUrl || '',
                            is_alliance: true,
                            prices: info.prices || []
                        };
                    });
                } else {
                    funeralHomesData = [];
                }"""

new_content = content.replace(target, replacement).replace(target2, replacement2)

with codecs.open('search.html', 'w', 'utf-8') as f:
    f.write(new_content)

print("Replacement successful")
